website/versioned_docs/version-5.x/server-side-rendering-overview.md

```diff
@@ -0,0 +1,294 @@
+
+In the context of microfrontends, server-side rendering refers to assembling the HTML from multiple, separate microfrontends. Each microfrontend controls a fragment of the HTML sent from web server to browser, and hydrate their fragment once initialized in the browser.
+
+## Purpose
+
+A primary purpose of server-side rendering is improved performance. Server rendered pages often display their content to users faster than their static counterparts, since the user is presented with the content before javascript resources have been initialized. Other reasons for SSR include improved search engine optimization (SEO).
+
+Server rendered applications are generally harder to build and maintain, since the code has to work on both client and server. Additionally, SSR often complicates the infrastructure needed to run your application, since many SPA + SSR solutions require NodeJS, which is not required in production for client-only SPAs.
+
+## Implementation Overview
+
+The ultimate goal of server-side rendering is to generate an HTTP response that the browser will display to the user while javascript is hydrating. Most microfrontend server-side rendering implementations, including single-spa's recommended approach, do this with the following steps:
+
+1. Layout - Identify which microfrontends to render for the incoming HTTP request, and where within the HTML they will be placed. This is usually route based.
+2. Fetch - Begin rendering the HTML for each microfrontend to a stream.
+3. Headers - Retrieve HTTP response headers from each microfrontend. Merge them together and send the result as the HTTP response headers to the browser.
+4. Body - Send the HTTP response body to the browser, which is an HTML document consisting of static and dynamic parts. This involves waiting for each microfrontend's stream to end before proceeding to the next portion of HTML.
+5. Hydrate - Within the browser, download all javascript needed to hydrate the HTML.
+
+## 1. Layout
+
+To define an HTML template that lays out your page, first choose a "microfrontend layout middleware":
+
+1. [single-spa-layout](/docs/layout-overview): The official layout engine for single-spa.
+2. [Tailor](https://github.com/zalando/tailor): A popular, battle tested layout engine that predates single-spa-layout and is not officially affiliated with single-spa.
+3. [TailorX](https://github.com/StyleT/tailorx): An actively maintained fork of Tailor that is used by [Namecheap](https://www.namecheap.com/) in their single-spa website. The single-spa core team collaborated with the creators of TailorX when authoring single-spa-layout, taking some inspiration from it.
+
+We generally recommend single-spa-layout, although choosing one of the other options might make sense for your situation, since single-spa-layout is newer and less battle tested than Tailor/TailorX.
+
+With single-spa-layout, you define a single template that handles all routes. [Full documentation](/docs/layout-definition).
```

Comment:
Fully self contained link text is a new accessibility concern for me to think about. Doing so would require changing hundreds of links across the documentation site, and I do not see the benefits. For both visual users and screen reader users, the text surrounding a link is crucial to understanding what the link is for and why it exists. Self contained link text makes the content read awkwardly in many cases, including this one.

I am not planning on making these changes - if I'm missing a reason why this is important, feel free to share more.

Reply:
>  Doing so would require changing hundreds of links across the documentation site

My suggestion was for just this one line, I don't think its necessary to jump to the extreme of having to change it for hundreds of links in one go. All others can be changed over time.

> For both visual users and screen reader users, the text surrounding a link is crucial to understanding what the link is for and why it exists. 

This is not always true, and most accessibility standards will disagree with you on this. For example: 

https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-refs.html
> The intent of this Success Criterion is to help users understand the purpose of each link so they can decide whether they want to follow the link. Whenever possible, provide link text that identifies the purpose of the link without needing additional context. Assistive technology has the ability to provide users with a list of links that are on the Web page. Link text that is as meaningful as possible will aid users who want to choose from this list of links. Meaningful link text also helps those who wish to tab from link to link. 

https://webaim.org/techniques/hypertext/
> Implication 1: Links should make sense out of context. Phrases such as "click here," "more," "click for details," and so on are ambiguous when read out of context. 

https://usability.yale.edu/web-accessibility/articles/links#disabilities
> While screen readers can read a full page to a user, screen reader users may prefer to instead listen to a list of links. In that case, a screen reader may only read the link text and not the surrounding text.

But do what you will. I tried to approach this as "not a big deal".

Reply:
>  I tried to approach this as "not a big deal".

Sounds like my previous comment maybe felt aggressive to you. I think with this pull request is a big deal for me because it is the culmination of ~6 months of (on and off) work on server-side rendering. I was hoping more feedback, and feedback related to the content, and think I got a bit frustrated seeing that there weren't many comments talking about the substance of the documentation here. This isn't your fault (as you gave some good comments in your review, and took the time to review it at all), and I should have reframed what I said about accessibility here. I appreciate your focus on accessibility.